### ISSUE TITLE
Feature: Add some sort of project scaffolding

### ISSUE THREAD

**Issue by JotraN** [ww-tech/roxie#5]
Roxie is pretty lightweight, but I'm also lazy so having to create different files each time I start up a new screen sometimes becomes a bit of copy/paste work from other screens.

Not sure how to go about this (probably just some shell script), but most likely it would generate these initial files for us:
- `Action`
- `Change`
- `State`
- `ViewModel`
- `ViewModelTest` 👍 👍 👍 

Possibly dagger related jazz too.

**Comment by AlecMedina** [comment]
Maybe something worth looking at is the Android Studio templates. I Just created one real quick like this 
<img width="865" alt="screen shot 2019-01-18 at 2 13 23 pm" src="https://user-images.githubusercontent.com/36166079/51408004-64313180-1b2b-11e9-8e9e-eeb00c8c8369.png">

and you can in the project view just choose the root and select the template. Not as automated as I would like but maybe we can do something with it. Food for thought.

**Comment by JotraN** [comment]
@AlecMedina, that feels slightly weird to me since we only need to generate the above files once per screen. 
Live templates seem like stuff we'd consistently type a lot.

Would definitely be helpful for tests though (https://medium.com/buffer-engineering/using-live-templates-to-write-tests-8d8041e390d3) - given how often we do the whole `Given x, when y, then z` format.

**Comment by AlecMedina** [comment]
This might be worth taking a look at as well: https://riggaroo.co.za/custom-file-template-group-android-studiointellij/. Having a custom wizard through the Android Studio IDE would be pretty nice. Whatever we implement should be user friendly-ish and I think ```file>new>MVI (or whatever we want to call it)``` and just fill in some details would be pretty nice.

**Comment by jshvarts** [comment]
@curioustechizen thanks for the suggestion! I think it does make sense to define `Action`, `Change` and `State` in the `ViewModel` class for that feature.

**Comment by rusili** [comment]
Oh I didn't know this was a thread. I've already built out a template for our MVI architecture, but it's very raw and not polished.
How would I share my folder?

**Comment by jshvarts** [comment]
Here is another good example https://proandroiddev.com/how-i-automated-creating-files-for-a-new-screen-with-my-own-android-studio-plugin-5d54b14ba6fa. It could be a separate module in this repo or a separate repo. Looking at these examples, there is a lot of code and tests to write and maintain--not sure if there is an immediate need for it.

**Comment by JotraN** [comment]
😮 that looks amazing.

I actually got around to implementing android studio templates - just haven't pushed it anywhere yet. I migggggght try to push something for review at some point in my life, but all I did was follow some medium article tutorial 😁. 

Essentially just had to create a couple files in AS plugins folder - so if anyone would want to use them, they'd have to copy it over to their folder themselves (this was also another reason why I haven't pushed them yet because I never got around to copying them over to my latest AS installation).
